alternator/CMakeLists.txt

```diff
@@ -25,11 +25,13 @@ target_include_directories(alternator
   PRIVATE
     ${RAPIDJSON_INCLUDE_DIRS})
 target_link_libraries(alternator
-  cql3
-  idl
-  Seastar::seastar
-  xxHash::xxhash
-  absl::headers)
+  PUBLIC
+    Seastar::seastar
+    xxHash::xxhash
```

Comment:
I'm trying to understand the difference between PULBIC and PRIVATE. The CMake documentation is laconic; I've found the following discussion on stackoverflow: https://stackoverflow.com/a/69786586

Based on that, I'm inclined to think that xxhash should be PUBLIC if the alternator lib header files referenced some xxhash symbols. The alternator header files that `main.cc` includes are `controller.hh` and `ttl.hh`. (Some test binaries include other alternator headers, too.) Either way, I've not managed to find any xxhash mentions in alternator.

Oh, wait, git-blame does explain it: see commit bd1ea104fe9c. It seems that we have some technical debt there:
~~~
also, `alternator` uses Seastar library and uses xxHash indirectly.
we should fix the latter by exposing the included header instead,
but for now, let's just link alternator directly to xxHash.
~~~


 Is that fixable, or very cumbersome?

Reply:
(I did verify that the externally used alternator headers expose seastar symbols, so seastar should indeed be PUBLIC.)

Reply:
> Is that fixable, or very cumbersome?

it's kind of cumbersome. because we include `xxhash.h` and define `XXH_PRIVATE_API` macro in order to inline the xxhash functions. there are two headers which are including `xxhash.h`:

- `utils/simple_hasher.hh`
- `utils/xx_hasher.hh`

both of them are under `utils`. and the `utils` target properly expose `xxHash::xxhash` with `PUBLIC`. in theory, whoever use either of these two headers should link against `utils` so that it can have access the `-DXXH_PRIVATE_API`. but i'd prefer leaving this to a follow-up change.

Reply:
Got it, thanks!

Reply:
So I'm fine with the patch being merged as-is.